### ISSUE TITLE
python-hpOneView BUG

### ISSUE THREAD

**Issue by bihary84** [HewlettPackard/python-hpOneView#238]
If I use amqp version 2.0 and just copy paste the code from github without any modification I got this trace:

```
[root@sanyichef ov]# ./scmb.py -a 192.168.x.x -u Administrator -p xxxxxxxx
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/amqp/connection.py", line 377, in channel
    return self.channels[channel_id]
KeyError: None

```

During handling of the above exception, another exception occurred:

```
Traceback (most recent call last):
  File "./scmb.py", line 199, in <module>
    sys.exit(main())
  File "./scmb.py", line 194, in main
    recv(args.host, args.route)
  File "./scmb.py", line 106, in recv
    ch = conn.channel()
  File "/usr/lib/python3.4/site-packages/amqp/connection.py", line 380, in channel
    channel.open()
  File "/usr/lib/python3.4/site-packages/amqp/channel.py", line 455, in open
    spec.Channel.Open, 's', ('',), wait=spec.Channel.OpenOk,
  File "/usr/lib/python3.4/site-packages/amqp/abstract_channel.py", line 62, in send_method
    conn._frame_writer.send((1, self.channel_id, sig, args, content))
AttributeError: 'Connection' object has no attribute '_frame_writer'
[root@sanyichef ov]#

If I upgrade the amqp from 2.0.0 to 2.1.4 I got this error message:
/usr/lib/python3.4/site-packages/amqp/connection.py:304: AMQPDeprecationWarning: The .frame_writer attribute on the connection was accessed before
the connection was established.  This is supported for now, but will
be deprecated in amqp 2.2.0.
Since amqp 2.0 you have to explicitly call Connection.connect()
before using the connection.

```
 
So, I have to added this line:
```
# Connect to RabbitMQ
    conn = amqp.Connection(dest, login_method='EXTERNAL', ssl=ssl_options)
    conn.connect() ###NEW LINE
    ch = conn.channel()

```
However after that I got this message:
```
Traceback (most recent call last):
  File "./scmb.py", line 199, in <module>
    sys.exit(main())
  File "./scmb.py", line 194, in main
    recv(args.host, args.route)
  File "./scmb.py", line 113, in recv
    ch.wait()
TypeError: wait() missing 1 required positional argument: 'method'

```
I was trying to find the description of the channel class but what I have found missed the wait() and callbacks() method. I have found java docs which explains this/similar methods but nothing for python. 
https://www.rabbitmq.com/amqp-0-9-1-reference.html
https://www.rabbitmq.com/releases/rabbitmq-java-client/v1.7.0/rabbitmq-java-client-javadoc-1.7.0/com/rabbitmq/client/Channel.html
 
 
Because it said a positional argument is missing I just added an argument ‘1’ and voilà it just started to work. And I don’t really understand this. Can you shed some light on this for me?
```
while ch.callbacks:
     ch.wait(1)

```

**Comment by GustavoHennig** [comment]
Hi, thank you for detailing the problem.

I tried to repeat your steps and I had the same problems.
Your suggestion to fix it was right.
On the last question, the wait signature was changed. I think this was a way that the developer who created the code chose to let the script listening. I believe it could be replaced by another code. However, as this code is no longer actively maintained, I just set the missing parameter.
The amqp uses some constants in this argument, so I set the constant `amqp.spec.Queue.BindOk`.

The fix is in this [PR](https://github.com/HewlettPackard/python-hpOneView/pull/240), I hope it will be merged soon.
